include/abb_librws/v1_0/rws_client.h

```diff
@@ -155,6 +155,17 @@ class RWSClient
                                        const std::string& tool = "",
                                        const std::string& wobj = "");
 
+  /**
+   * \brief Checks progress of provided resource.
+   * 
+   * \param id Resource id to check
+   *
+   * \return \a RWSResult containing the result.
+   *
+   * \throw \a RWSError if something goes wrong.
+   */
+  RWSResult getProgress(std::string const& id);
```

Comment:
@mkatliar [you mentioned](https://github.com/NoMagicAi/abb_librws/pull/48#discussion_r936494588)
> Should not be a member function, as it does not require access to private members of `RWSClient`.

Should it be moved to a separate source file like the [retcode implementation](https://github.com/NoMagicAi/abb_librws/pull/49/files)?